src/schema/schema_create.c

```diff
@@ -980,6 +980,23 @@ __schema_create(WT_SESSION_IMPL *session, const char *uri, const char *config)
         WT_RET_MSG(session, ENOTSUP,
           "%s: import is only supported for 'file' and 'table' data sources", uri);
 
+    /*
+     * If tiered storage is configured at the connection level and the user has not configured
+     * tiered_storage.name to be none, then the object being created is a tiered object.
+     */
+    is_tiered = S2C(session)->bstorage != NULL &&
+      !((ret = __wt_config_getones(session, config, "tiered_storage.name", &cval)) == 0 &&
+        cval.len != 0 && WT_STRING_MATCH("none", cval.str, cval.len));
+
+    /*
+     * If the type configuration is set to anything else but "file" while using tiered storage we
+     * must fail the operation.
+     */
+    if (is_tiered && (ret = __wt_config_getones(session, config, "type", &cval)) == 0 &&
+      !WT_STRING_MATCH("file", cval.str, cval.len))
+        WT_RET_MSG(
+          session, ENOTSUP, "unsupported type configuration: type must be file for tiered storage");
+
```

Comment:
Minor note: I think that it would now be cleaner to move all the configuration checking to something like `schema_create_config_chk` as that function is now looking more complex and it would keep all the config checking in one contained place.

Reply:
Good point, I'll move the config checking into another function.